internal/component/redis/redis.go

```diff
@@ -37,110 +79,71 @@ func ParseClientFromProperties(properties map[string]string, defaultSettings *Se
-		opts.SentinelAddrs = strings.Split(s.Host, ",")
-
-		return redis.NewFailoverClusterClient(opts)
-	}
-
-	return redis.NewFailoverClient(opts)
 }
 
-func newClient(s *Settings) redis.UniversalClient {
-	if s == nil {
-		return nil
+func ClientHasJSONSupport(c RedisClient) bool {
+	if clientHasJSONSupport != nil {
+		return *clientHasJSONSupport
 	}
-	if s.RedisType == ClusterType {
-		options := &redis.ClusterOptions{
-			Addrs:              strings.Split(s.Host, ","),
-			Password:           s.Password,
-			Username:           s.Username,
-			MaxRetries:         s.RedisMaxRetries,
-			MaxRetryBackoff:    time.Duration(s.RedisMaxRetryInterval),
-			MinRetryBackoff:    time.Duration(s.RedisMinRetryInterval),
-			DialTimeout:        time.Duration(s.DialTimeout),
-			ReadTimeout:        time.Duration(s.ReadTimeout),
-			WriteTimeout:       time.Duration(s.WriteTimeout),
-			PoolSize:           s.PoolSize,
-			MaxConnAge:         time.Duration(s.MaxConnAge),
-			MinIdleConns:       s.MinIdleConns,
-			PoolTimeout:        time.Duration(s.PoolTimeout),
-			IdleCheckFrequency: time.Duration(s.IdleCheckFrequency),
-			IdleTimeout:        time.Duration(s.IdleTimeout),
-		}
-		/* #nosec */
-		if s.EnableTLS {
-			options.TLSConfig = &tls.Config{
-				InsecureSkipVerify: s.EnableTLS,
-			}
-		}
+	err := c.DoErr(context.Background(), "JSON.GET")
 
-		return redis.NewClusterClient(options)
+	if err == nil {
+		trueVal := true
+		clientHasJSONSupport = &trueVal
+		return true
 	}
 
-	options := &redis.Options{
-		Addr:               s.Host,
-		Password:           s.Password,
-		Username:           s.Username,
-		DB:                 s.DB,
-		MaxRetries:         s.RedisMaxRetries,
-		MaxRetryBackoff:    time.Duration(s.RedisMaxRetryInterval),
-		MinRetryBackoff:    time.Duration(s.RedisMinRetryInterval),
-		DialTimeout:        time.Duration(s.DialTimeout),
-		ReadTimeout:        time.Duration(s.ReadTimeout),
-		WriteTimeout:       time.Duration(s.WriteTimeout),
-		PoolSize:           s.PoolSize,
-		MaxConnAge:         time.Duration(s.MaxConnAge),
-		MinIdleConns:       s.MinIdleConns,
-		PoolTimeout:        time.Duration(s.PoolTimeout),
-		IdleCheckFrequency: time.Duration(s.IdleCheckFrequency),
-		IdleTimeout:        time.Duration(s.IdleTimeout),
+	if strings.HasPrefix(err.Error(), "ERR unknown command") {
+		falseVal := false
+		clientHasJSONSupport = &falseVal
+		return false
 	}
+	trueVal := true
+	clientHasJSONSupport = &trueVal
+	return true
+}
 
-	/* #nosec */
-	if s.EnableTLS {
-		options.TLSConfig = &tls.Config{
-			InsecureSkipVerify: s.EnableTLS,
+func GetServerVersion(c RedisClient) (string, error) {
+	ctx := c.Context()
+	res, err := c.DoResult(ctx, "INFO", "server")
+	if err != nil {
+		return "", err
+	}
+	// get row in string res beginning with "redis_version"
+	rows := strings.Split(res.(string), "\n")
+	for _, row := range rows {
+		if strings.HasPrefix(row, "redis_version:") {
+			return strings.TrimSpace(strings.Split(row, ":")[1]), nil
```

Comment:
Nit: strings.Cut ? In case there are more `:` after the first=

Reply:
Not possible given the output returned by this Redis command.